packages/unlock-js/src/web3Service.ts

```diff
@@ -939,4 +939,57 @@ export default class Web3Service extends UnlockService {
     const unlockContract = await this.getUnlockContract(unlockAddress, provider)
     return await unlockContract.publicLockLatestVersion()
   }
+
+  /**
+   * Returns true if the address is a valid EOA
+   */
+
+  async isValidEOA(address: string) {
+    const provider = this.providerForNetwork(1)
+    const code = await provider.getCode(address)
+    const count = await provider.getTransactionCount(address)
+    if (code === '0x' && count > 0) {
+      return true
+    } else {
+      return false
+    }
+  }
+
+  /**
+   * Returns the resolved ens address that's a valid EOA
+   */
+  async resolveEns(ensName: string) {
+    const provider = this.providerForNetwork(1)
+    try {
+      const ensAddress = await provider.resolveName(ensName)
+      if (ensAddress !== null) {
+        const isValid = await this.isValidEOA(ensAddress)
+        if (isValid === true) {
+          return ensAddress
+        } else throw 'The ens address is not a valid EOA'
```

Comment:
Throw an instance of Error instead of string. That way, someone can check if instance of error and also get other properties attached to an error. 

Same below.

Reply:
Cool! Do you recommend throwing it like this: `throw new Error('The ens name is not a valid EOA')`? @searchableguy

Reply:
Yes.

Reply:
Cool, thanks!